### ISSUE TITLE
CSS variables incorrectly optimized using prod flag

### ISSUE THREAD

**Issue by David-Development** [angular/angular-cli#15684]
# 🐞 Bug report

### Command (mark with an `x`)

```
- [ ] new
- [x] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Description

When defining border-radius property for each side manually (using `border-top-left-radius / border-bottom-left-radius / ...`), `ng build --prod` will merge those definitions together into a single `border-radius: ...` definition. However when using variables while still providing fallback values for older browsers who don't support variables, the variables will be ignored when using `ng build --prod`. This only happens when using the flag `--prod`.

## 🔬 Minimal Reproduction

For reproduction purposes I created a simple Angular App.

```bash
git clone https://github.com/David-Development/angular-css-minify-bug-app
cd angular-css-minify-bug-app
npm install
ng serve 
```

You'll notice that both divs look the same (both have rounded edges on the left side). So far, everything is correct.

However, after running `ng build --prod`, you'll notice, that the second div has no round edges on the left side anymore. Somehow (I guess by doing some minifying) angular will merge the four attributes (`border-top-left-radius / border-bottom-left-radius / border-top-right-radius / border-bottom-right-radius`) together. The problem is, that by merging those attributes together, it is not considering variables.

## 🔥 Exception or Error
The following css code: 
<pre><code>
--control-border-radius: 10px;

border-top-left-radius: 1px;
border-bottom-left-radius: 1px;
    
border-top-left-radius: var(--control-border-radius);
border-bottom-left-radius: var(--control-border-radius);
border-top-right-radius: 0px;
border-bottom-right-radius: 0px;
</code></pre>

is being converted to: (using `ng build --prod`)

<pre><code>
--control-border-radius: 10px;
...
border-top-left-radius: 1px;
border-bottom-left-radius: 1px;
    
border-top-left-radius: var(--control-border-radius);
border-bottom-left-radius: var(--control-border-radius);
border-top-right-radius: 0px;
border-bottom-right-radius: 0px;
border: 1px 0px 0px 1px;
</code></pre>

## 🌍 Your Environment
Tested in Angular 7 and 9

## Related issue:
- https://github.com/angular/angular-cli/issues/10579

Unfortunately the described workaround does not work for me.
```
/* clean-css ignore:start */
...
/* clean-css ignore:end */
```

**Comment by alan-agius4** [comment]
Hi @David-Development, I tried your reproduction and was able to use the clean-css ignore comments successfully.

```css
div.test2 {
    background: #ff0000a8;

    /* clean-css ignore:start */
    border-top-left-radius: 1px;
    border-bottom-left-radius: 1px;

    border-top-left-radius: var(--control-border-radius);
    border-bottom-left-radius: var(--control-border-radius);
    border-top-right-radius: 0px;
    border-bottom-right-radius: 0px;
    /* clean-css ignore:end */
}
```

![image](https://user-images.githubusercontent.com/17563226/65690778-0a94e580-e070-11e9-9c82-93a334ff587f.png)

**Comment by David-Development** [comment]
Hi @alan-agius4, thanks for your fast response. I ran my tests again using the `clean-css ` flags as you described above and now it works for me too. Not sure what went wrong when I tried it the first time. And thanks for your example!

My follow up question would be: Is that something that should be fixed? The workaround is working fine but I feel like this could save devs many hours of digging into why your application looks differently in `production` mode than it looks like in `dev` mode.
